### DISCUSSION TITLE
SAFEBOOT Shelly Plus 1 Tasmota Upload fehlgeschlagen  Nicht genug Speicherplatz

### DISCUSSION THREAD

**Discussion by LeoZanker** [arendst/Tasmota#21700] [Q&A]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ ] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [ ] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [ ] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [ ] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): _____
- [ ] Tasmota binary firmware version number used: _____
  - [ ] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by Jason2866** [comment]
See #18983

**Comment by Reddingo** [comment]
I have the same issue upgrading my Shelly 1Plus PM. Pointing to an old thread with different preconditions did not help at all.
I started with my process on 12.5, used the Partition_Wizard to migrate to 

> Program Version	14.1.0 (release-safeboot) single-core

and ended up with this layout:

> ESP Chip Id	679500 (ESP32-U4WDH-D v3.0)
Flash Chip Id	0x164020 (DIO)
Flash Size	4096 KB
Program Flash Size	4096 KB
Program Size	812 KB
Free Program Space	1536 KB
Free Memory	211.3 KB (frag. 42%)

> Partition app_0*	1536 KB (used 52%)
Partition fs	448 KB
Partition app_1	1536 KB
Partition fs	448  KB


Any hints what to do now? Being in this SAFEBOOT mode I cannot upload any Partition_Wizard, **Restart 3** (seen in the other threads) does not work. I also uploaded the **tasmota32solo1-safeboot.bin** two times but the storage layout remains the same.

**Reply by Jason2866** [reply]
Download this version [solo1](http://ota.tasmota.com/tasmota32/release-12.5.0/tasmota32solo1.bin) (newer ones are too big!) and upload via firmware upgrade by file. Do this TWO!! times. Since it seems you have uploaded safeboot firmware manually. Uploading safeboot is NEVER necessary when doing the Shelly OTA process as described. After that do the Autoconfiguration again. Now you should be able to start the conversion process.

**Reply by Reddingo** [reply]
I did what you proposed: 
Downloaded the 12.5 solo1
Did the Firmware-Update with the same file twice.

![grafik](https://github.com/arendst/Tasmota/assets/32964022/11a876ce-b897-4374-9913-4eb453e5270d)

Results still in:
> Program Version	14.1.0 (release-safeboot) single-core

and
> Partition app_0*	1536 KB (used 52%)
Partition fs	448 KB
Partition app_1	1536 KB
Partition fs	448 KB

**Reply by Jason2866** [reply]
Good, uploading has worked. But still starting always in the same OTA slot. Fire the command `restart 3`
after the device is back, do the upgrade with the downloaded solo1 firmware again. Maybe you have to try a few times. If you are in normal Tasmota Firmware start the Autoconfig. The Autoconfig Step is explained in the Readme of the OTA Shelly Convert process.

Not following exactly the Shelly OTA process as described will end in issues! The warning makers are there for good reasons!

**Reply by Reddingo** [reply]
There is no valid option for restart #3

>16:59:38.257 CMD: restart 3
16:59:38.266 MQT: stat/tasmota_ShellyPlusPM1_2/RESULT = {"Command":"Error"}
16:59:45.486 CMD: restart
16:59:45.493 MQT: stat/tasmota_ShellyPlusPM1_2/RESULT = {"Restart":"1 to restart, 2 to halt"}

**Reply by Jason2866** [reply]
Okay, so serial flash is the only way.

Edit: Theo just tested. The command `restart 3` is there in safeboot firmware and switching the OTA slot. Idk why it is not a valid command in your case??

EDIT2: Did you have refreshed the Web page? Maybe you are already in solo1 firmware and only the old website is shown! Firmware 12.5 does not have this command and would explain the behaviour.
